service/controller/chartconfig/v7/resource/chart/create.go

```diff
@@ -6,16 +6,30 @@ import (
 	"fmt"
 
 	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/resourcecanceledcontext"
 	"k8s.io/helm/pkg/helm"
 
-	"github.com/giantswarm/chart-operator/service/controller/chartconfig/v6/key"
+	"github.com/giantswarm/chart-operator/service/controller/chartconfig/v7/key"
 )
 
 func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange interface{}) error {
 	customObject, err := key.ToCustomObject(obj)
 	if err != nil {
 		return microerror.Mask(err)
 	}
+
+	reason, reasonOk := customObject.Labels[key.CordonReasonAnnotationName]
+	until, untilOk := customObject.Labels[key.CordonUntilAnnotationName]
+
+	if reasonOk && untilOk {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("chart %#q had been cordoned off until %s with following reason; %s ", key.ChartName(customObject), until, reason))
```

Comment:
It will print the message like, ```chart `quay.io/giantswarm/chart-operator-chart` had been cordoned off until 2019-12-31T23:59:59Z with following reason; testing upgrade "```

Reply:
Message is good but I'd just suggest a couple of changes.

"chart `coredns` had been cordoned until 2019-12-31T23:59:59Z due to reason `testing upgrade `"

Please also add `key.CordonReason` and `key.CordonUntil` for getting the values.